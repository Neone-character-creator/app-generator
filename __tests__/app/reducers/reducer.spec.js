const reducer = require("../../../project-template/src/main/resources/scripts/reducer").default;
describe("state reducer", function(){
    it("throws an error if an invalid action type is passed with an existing state", function(){
       expect(()=>{
           reducer({}, {
               type: "FOO"
           });
       }).toThrowErrorMatchingSnapshot();
    });
    it("throws an error if the path does not begin with $state", function(){
        expect(()=>{
            reducer({}, {
                type: "SET",
                path: "foobar"
            });
        }).toThrowErrorMatchingSnapshot();
    });
    it("creates a copy of the previous state", function(){
        const initialState = {
            transformers: []
        };
        const outputState = reducer(initialState, {
            type: "SET",
            path: "$state.foo"
        });
        expect(initialState).not.toEqual(outputState);
    });
});