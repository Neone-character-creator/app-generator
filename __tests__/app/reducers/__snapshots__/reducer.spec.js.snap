// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state reducer throws an error if an invalid action type is passed with an existing state 1`] = `"Action type FOO is not supported."`;

exports[`state reducer throws an error if the path does not begin with $state 1`] = `"targetPath must not be null or undefined"`;
